### PR TITLE
SQL Query Generation from Cron Tick

### PR DESCRIPTION
**Pull request by brendoncarroll**
Adds a transform `sql-gen-queries` to `pachtf` which reads cron timestamp files from the input repo, and generates SQL queries with the timestamp as a comment in the output repo.

This is a component of a SQL ingest DAG, it can be replaced with something more complicated to allow for custom sharding, or snapshotting the database table before ingest.

### CODE REVIEW

File: Dockerfile.pachtf [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM scratch
+FROM alpine:3.15
```
Comment (acohen4) [inline Dockerfile.pachtf:1]: why the image swap?

File: src/internal/transforms/sqlingest.go [hunk 2]
```diff
@@ -96,10 +100,48 @@ func makeWriterFactory(formatName string) (writerFactory, error) {
 		}
 	case "csv":
 		factory = func(w io.Writer, fieldNames []string) sdata.TupleWriter {
-			return sdata.NewCSVWriter(w, fieldNames)
+			return sdata.NewCSVWriter(w, nil)
```
Comment (acohen4) [inline src/internal/transforms/sqlingest.go:103]: is this change intentional?

File: src/internal/transforms/sqlingest.go [hunk 3]
```diff
@@ -96,10 +100,48 @@ func makeWriterFactory(formatName string) (writerFactory, error) {
...
+
+// SQLQueryGeneration generates queries with a timestamp in the comments
+func SQLQueryGeneration(ctx context.Context, params SQLQueryGenerationParams) error {
+	timestamp, err := readCronTimestamp(params.Logger, params.InputDir)
+	if err != nil {
+		return err
+	}
+	timestampComment := fmt.Sprintf("-- %d\n", timestamp)
+	contents := timestampComment + params.Query
+	outputPath := filepath.Join(params.OutputDir, "0000")
```
Comment (acohen4) [inline src/internal/transforms/sqlingest.go:126]: I'm missing something here. why "pfs/out/0000" ?

File: src/server/cmd/pachtf/main.go [hunk 4]
```diff
@@ -26,14 +26,20 @@ func main() {
...
 		log.Fatalf("unrecognized transform name %q", transformName)
 	}
+	if err := entrypoint(ctx, log, transformArgs); err != nil {
+		log.Fatal(err)
+	}
+}
+
+type Entrypoint = func(ctx context.Context, log *logrus.Logger, args []string) error
+
+var entrypoints = map[string]Entrypoint{
```
Comment (acohen4) [inline src/server/cmd/pachtf/main.go:40]: 👍

File: Dockerfile.pachtf (hunk 1 above)
Comment (brendoncarroll) [reply Dockerfile.pachtf:1]: We assume a temp directory exists in the worker, and this was causing the pipeline to break.  I wasn't sure what else we needed so I just switched to a real base image instead.

File: src/internal/transforms/sqlingest.go (hunk 2 above)
Comment (brendoncarroll) [reply src/internal/transforms/sqlingest.go:103]: Yeah, setting the csv writer fields to nil removes the header.  I don't really see the benefit of the header since the user will know what the asked for and it makes splitting and concatenating files more difficult.

File: src/internal/transforms/sqlingest.go (hunk 3 above)
Comment (brendoncarroll) [reply src/internal/transforms/sqlingest.go:126]: `0000` is just the path that the query generation writes to by default.  I don't want users to get accustomed to something like RESULTS.json.  They should be writing code to consume an unknown number of files from the output, which will be based on how many separate queries they asked for in the input.  I'm open to suggestions here, I was thinking you would just increment this number if we ever start generating more than one query.

File: src/internal/transforms/sqlingest.go (hunk 2 above)
Comment (acohen4) [reply src/internal/transforms/sqlingest.go:103]: makes sense 👍

File: src/server/cmd/pachtf/main.go [hunk 5]
```diff
@@ -75,8 +90,14 @@ func sqlGenQueries(ctx context.Context, log *logrus.Logger, args []string) error
 		return errors.Errorf("must provide query")
 	}
 	query := args[0]
-	inputDir := filepath.FromSlash(pfs + "/in")
-	outputDir := filepath.FromSlash(pfsOut)
+	inputDir, err := filepath.EvalSymlinks(filepath.FromSlash(pfs + "/in"))
```
Comment (acohen4) [inline src/server/cmd/pachtf/main.go:93]: what's the case where these paths would need symlinks resolved?

**Review by acohen4 — APPROVED** [review]
LGTM

File: src/server/cmd/pachtf/main.go (hunk 5 above)
Comment (brendoncarroll) [reply src/server/cmd/pachtf/main.go:93]: The input subdirectories (the immediate children of `/pfs/`) are symlinks.  `filepath.WalkDir` doesn't follow symlinks.
